packages/orchestrator/main.go

```diff
@@ -18,28 +23,87 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	port := flag.Int("port", defaultPort, "orchestrator server port")
+	sig, sigCancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer sigCancel()
 
+	var port uint
+
+	flag.UintVar(&port, "port", defaultPort, "orchestrator server port")
 	flag.Parse()
 
+	wg := &sync.WaitGroup{}
+	exitCode := &atomic.Int32{}
+	telemetrySignal := make(chan struct{})
```

Comment:
this isn't used, right? Something is probably missing

Reply:
we block on it in 47 and close it in 98... just so that telemetry is the last thing to shut down.